dist/js/dnc.js

```diff
@@ -0,0 +1,398 @@
+    **  to make it more testable and think about the
+    **  idea plugin idea mentioned in above TODO(s)
+    ** 
+    */
+    ops :  {
+      // main execution for operations
+      execute: function(newLayer) {
+        // TODO: another tight coupling between map
+        DNC.map.addLayer(newLayer, newLayer.geometry, DNC.map.numLayers);
+      },
+
+      // all geometry processes
+      geom: {
+        buffer: function(object, info) {
+          console.log(object, info);
+          var newLayer = {
+            geometry: turf.buffer(object, 0.1),
+            name: 'buffer_' + info.name
+          };
+
+          return newLayer;
+        },
+
+        // var union = turf.union(poly1, poly2);
+        union: function(object1, object2, info1, info2) {
+
+          var poly1 = object1,
+              poly2 = object2,
+              info1Strip = info1.name.replace('.geojson', ''),
+              info2Strip = info2.name.replace('.geojson', '');
+
+          if (object1.features) poly1 = object1.features[0];
+          if (object2.features) poly2 = object2.features[0];
+
+          var newLayer = {
+            geometry: turf.union(poly1, poly2),
+            name: 'union_' + info1Strip + '_' + info2Strip + '.geojson'
+          };
+          return newLayer;
+        },
+
+        erase: function(object1, object2, info1, info2) {
+          var poly1 = object1,
+              poly2 = object2,
+              info1Strip = info1.name.replace('.geojson', ''),
+              info2Strip = info2.name.replace('.geojson', '');
+
+          var newLayer = {
+            geometry: turf.erase(poly1, poly2),
+            name: 'erase_' + info1Strip + '_' + info2Strip + '.geojson'
+          };
+          return newLayer;
+
+        }
+      }
+    }
+
+};
+
+// NOTE: we are returning a class, not an instance
+module.exports =  Menu;
+
+},{}],4:[function(require,module,exports){
+'use strict';
+var Dropzone = require( '../dropzone' ) ,
+    GeoMenu = require( '../geoprocessing_menu' );
+
+var Map = function( options ) {
+    this.options = options || {}; 
+    if ( !( this instanceof Map ) ) { 
+        return new Map();
+    }
+
+    /*
+    **
+    **  TODO: these variables should be put 
+    **  into properties on the prototype
+    **  see the getters/setters section below
+    **
+    */
+    this.numLayers = 0;
+    this.map = null;
+    this.selection = {
+        add: function(l) {
+            this.list.push(l);
+        },
+        remove: function(l) {
+            for (var i = 0; i < this.list.length; i++) {
+              if (l._leaflet_id == this.list[i].layer._leaflet_id) {
+                this.list.splice(i, 1);
+              }
+            }
+        },
+        list: []
+    };
+    this.download = document.getElementById('download');
+
+};
+
+Map.prototype = {
+
+    init : function() {
+
+        // wireup the other classes 
+        DNC.dropzone = new Dropzone()
+        DNC.menu = new GeoMenu()
```

Comment:
I'm not super sharp on auto semicolon insertion. Is this valid?

On a style level, should it be allowed?